File: bundle/base/apps/demo/server/services/samples/javascript/query-parameter.js
/**
 *
 *  EN: EXECUTE A QUERY WITH PARAMETERS AND RETURN THE RESULT AS JSON
 *
 *  PT: EXECUTA UMA QUERY COM PARÂMETROS E RETORNA O RESULTADO COMO JSON
 *
 */
import {_db, _out, _req} from "@netuno/server-types";

let tableName = 'worker';
let columnName = 'name';

if (_db.config().getString('name') === 'demo_pt') {
  tableName = 'trabalhador';
  columnName = 'nome';
}

_out.json(
  _db.query(
    `SELECT uid, ${columnName} as "name"
    FROM ${tableName}
    WHERE id = ?::int AND ${columnName} like '%' || ?::varchar || '%'
    ORDER BY ${columnName}`,
    _req.getInt("id"),
    _req.getString("name")
  )
);

File: bundle/base/apps/demo/server/services/samples/ruby/query-parameter.rb

#
#  EN: EXECUTE A QUERY WITH PARAMETERS AND RETURN THE RESULT AS JSON
#
#  PT: EXECUTA UMA QUERY COM PARÂMETROS E RETORNA O RESULTADO COMO JSON
#

tableName = "worker"
columnName = "name"

if _db.config().getString("name") == "demo_pt" then
    tableName = "trabalhador"
    columnName = "nome"
end

_out.json(
    _db.query(%{
            SELECT *
            FROM #{tableName}
            WHERE id = ?::int AND #{columnName} LIKE \'%\' || ?::varchar || \'%\'
            ORDER BY #{columnName}
        },
        _req.getInt("id"),
        _req.getString("name")
    )
)
